service/history/handler.go

```diff
@@ -551,12 +550,14 @@ func (h *Handler) RespondDecisionTaskFailed(
 		token.RunID,
 		token.ScheduleID))
 
-	if failedRequest != nil && failedRequest.GetCause() == workflow.DecisionTaskFailedCauseUnhandledDecision {
-		h.GetLogger().Info("Non-Deterministic Error", tag.WorkflowRunID(token.RunID))
+	if failedRequest != nil && failedRequest.GetCause() == gen.DecisionTaskFailedCauseUnhandledDecision {
+		h.GetLogger().Info("Non-Deterministic Error", tag.WorkflowDomainID(token.DomainID), tag.WorkflowID(token.WorkflowID), tag.WorkflowRunID(token.RunID))
 		domainName, err := h.GetDomainCache().GetDomainName(token.DomainID)
+		counter := metrics.CadenceErrNonDeterministicCounter
```

Comment:
Maybe metricName is more proper than counter here? since `counter` make people think it's a number.

Reply:
I would suggest create a variable for domainTag and set different for it based on the error returned. Then you can write  `h.GetMetricsClient().Scope(scope, tag).IncCounter(metrics.CadenceErrNonDeterministicCounter)` only once. Just my personal opinion. Your call.